docs/blog/contributor-development-maintenance-guide.en-US.md

```diff
@@ -0,0 +1,155 @@
+
+2. In `package.json`, for the dependencies that we want to update when there is a new patch version, we use `~` description version number to allow patch version update.
+
+```json
+{
+  "dependencies": {
+    "rc-cascader": "~3.9.0"
+  }
+}
+```
+
+For the difference between `^` and `~` in the version description of package.json, see [What's the difference between tilde(~) and caret(^) in package.json](https://stackoverflow.com/questions/22343224/whats-the-difference-between-tilde-and-caret-in-package-json).
+
+Thus, when our dependency such as rc-cascader fixes a bug and releases a patch version such as: `3.9.1`, then the user's latest installed version is `3.9.1`. For the maintainer of Ant Design, we only need to execute the following command:
+
+```bash
+git clean -fdx
+npm i
+```
+
+### Snapshot update problem
+
+In Ant Design, we use Jest for unit test. Many first-time contributors of Ant Design may be confused that CI failed after opening a pull request in which they just modify the text in a demo. This starts with Ant Design's snapshot test.
+
+In most libraries, the concept of idempotence is emphasized, meaning that the same method, no matter how many times I execute the line, as long as the input parameters are the same, then the result will be the same.In Ant Design, I think the biggest use of snapshot detection is to verify the idempotence of our demo, so as to ensure the stability and certainty of component output.In fact, the principle of snapshot detection is very simple, is to save our demo generated html string, the next time in the running of the test task for comparison, if there is a difference that indicates that the snapshot detection failed.
+
+Going back to the original question, what should we do if we change a demo and find that the snapshot detection fails?
+
+1. First of all, we need to check the inconsistency of snapshot comparison to see if it is the expected change. If only the text is changed, and only the text is different from the snapshot comparison result, we just need to run the following command to update the snapshot:
+
+   ```bash
+   # Run the test command to update snapshot
+   npm run test -u
+   ```
+
+2. However, if the changes during snapshot comparison exceed the scope of your modification this time, for example, you only changed the text, but found that classnames in snapshots have changed, which obviously does not meet expectations, we need to rectify the causes.
+
+   The common causes are as follows:
+
+   - Local dependency is too old, it is possible to pull the latest code, but did not update the local dependency, resulting in the dependency package version is too low and the output results are inconsistent.The
+
+     solution is as simple as deleting the lock file、node_modules，and reinstall dependencies.
+
+   - You may have locally changed the core code in addition to the demo, causing the logic to change. You need to go over this change carefully.
```

Comment:
```suggestion
   - You may have locally changed the source code not only in demos,  which would cause unexpected change. You need to go through your change carefully.
```